cmake/kokkos_functions.cmake

```diff
@@ -6,7 +6,12 @@
 # upper-case version for use within
 
 set(Kokkos_OPTIONS_NOT_TO_EXPORT
-  Kokkos_ENABLE_TESTS  Kokkos_ENABLE_EXAMPLES)
+  Kokkos_ENABLE_BENCHMARKS
+  Kokkos_ENABLE_EXAMPLES
+  Kokkos_ENABLE_TESTS
+  Kokkos_ENABLE_HEADER_SELF_CONTAINMENT_TESTS
+  Kokkos_ENABLE_COMPILER_WARNINGS
+)
```

Comment:
I would probably add `DEPRECATION_WARNINGS` if you want to `Kokkos_ENABLE_CUDA_RELOCATABLE_CODE` anyway.

Reply:
The deprecation warnings are different.  Although I am not convinced it is really useful, in contrast to the ones I listed, it is something that is observable downstream.

Reply:
Sure. I can see that `KOKKOS_ENABLE_DEPRECATED_CODE_*` might be relevant but `DEPRECATION_WARNINGS` could also be an impl variable.